src/main/java/de/javakaffee/kryoserializers/guava/UnmodifiableNavigableSetSerializer.java

```diff
@@ -0,0 +1,70 @@
+package de.javakaffee.kryoserializers.guava;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.Serializer;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableSortedSet;
+import com.google.common.collect.Sets;
+
+import java.lang.reflect.Field;
+import java.util.NavigableSet;
+import java.util.TreeSet;
+
+/**
+ * A kryo {@link Serializer} for guava-libraries {@link ImmutableSortedSet}.
+ */
+public class UnmodifiableNavigableSetSerializer extends Serializer<NavigableSet<?>> {
+
+    @VisibleForTesting
+    protected static Object getDelegateFromUnmodifiableNavigableSet(NavigableSet<?> object) {
+        try {
+            Class<?> clazz = Class.forName(Sets.class.getCanonicalName() + "$UnmodifiableNavigableSet");
+            Field field = clazz.getDeclaredField("delegate");
+            field.setAccessible(true);
```

Comment:
This field should be resolved and cached at construction time so that it's not done per write.